### PR TITLE
Added product type into the order payload for PDP checkout

### PR DESCRIPTION
**Pull request by KyerdnA**
Added product type into the order payload for PDP checkout

Fixes: (link ticket)

#changelog Added product type into the order payload for PDP checkout

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202482535614265

### CODE REVIEW

File: view/frontend/templates/button_product_page.phtml [hunk 1]
```diff
@@ -289,23 +289,35 @@ foreach ($block->getAdditionalCheckoutButtonAttributes() as $attrName => $attrVa
                 <?php if ($block->isGrouped()): ?>
                     <?php foreach ($block->getGroupedProductChildren() as $childProduct):?>
                         var childProductQty = getGroupedProductChildQty(<?= /* @noEscape */ $childProduct->getId(); ?>);
+                        <?php if ($childProduct->getTypeInstance()->isVirtual($childProduct)): ?>
```
Comment (vitaliyreznikov) [inline view/frontend/templates/button_product_page.phtml:292]: Let's calculate productType only if childProductQty > 0, it increases readability

File: view/frontend/templates/button_product_page.phtml [hunk 2]
```diff
@@ -289,23 +289,35 @@ foreach ($block->getAdditionalCheckoutButtonAttributes() as $attrName => $attrVa
...
                                 name: '<?= /* @noEscape */ addcslashes($childProduct->getName(), "'"); ?>',
                                 quantity: childProductQty,
-                                options: JSON.stringify(options)
+                                options: JSON.stringify(options),
+                                type: productType
                             });
                         }
                     <?php endforeach; ?>
                 <?php else: ?>
+                    <?php if ($block->getProduct()->getTypeInstance()->isVirtual($block->getProduct())): ?>
```
Comment (vitaliyreznikov) [inline view/frontend/templates/button_product_page.phtml:309]: Why do we need this code and not simple $block->getProduct()->isVirtual() ?

**Comment by danac-gs** [comment]
@KyerdnA please follow the procedure and add full context to the PR description, e.g. link to the task.

File: view/frontend/templates/button_product_page.phtml [hunk 3]
```diff
@@ -290,22 +290,34 @@ foreach ($block->getAdditionalCheckoutButtonAttributes() as $attrName => $attrVa
...
                                 name: '<?= /* @noEscape */ addcslashes($childProduct->getName(), "'"); ?>',
                                 quantity: childProductQty,
-                                options: JSON.stringify(options)
+                                options: JSON.stringify(options),
+                                type: productType
                             });
                         }
                     <?php endforeach; ?>
                 <?php else: ?>
+                    <?php if ($block->getProduct()->isVirtual($block->getProduct())): ?>
```
Comment (vitaliyreznikov) [inline view/frontend/templates/button_product_page.phtml:309]: Just to better understand underlying magento code, why do we need `$product->isVirtual($product)` and not `$product->isVirtual()`

Comment (KyerdnA) [reply view/frontend/templates/button_product_page.phtml]: In this case i've just missed to removed `$product` parameter. 
It won't break anything, but it's unneeded things in the code.
The code that was here before - `$product->getTypeInstance()->isVirtual($product)` - is how it's actually implemented under the hood of isVirtual() function. But obviously we don't need that here as we have just a simple isVirtual() wrapper. 

It's removed with last commit.

**Review by vitaliyreznikov — APPROVED** [review]

**Pull request merged**